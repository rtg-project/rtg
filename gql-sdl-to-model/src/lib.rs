pub mod conversion_error;
pub mod convert_document;
pub mod convert_graphql_string;
pub mod convert_type_definition;

// Tests
#[cfg(test)]
mod tests {

  use super::*;
  use insta::assert_debug_snapshot;

  #[test]
  fn it_works() {
    let model = convert_graphql_string::convert_graphql_string(
      r#"
      type toto {
        a: id
        b: int
      }
      "#,
    )
    .unwrap();
    assert_debug_snapshot!(model);
  }
}

// Test the code in the readme file
// See https://github.com/rust-lang/cargo/issues/383#issuecomment-720873790
#[cfg(doctest)]
mod test_readme {
  macro_rules! external_doc_test {
    ($x:expr) => {
      #[doc = $x]
      extern "C" {}
    };
  }

  external_doc_test!(include_str!("../README.md"));
}
