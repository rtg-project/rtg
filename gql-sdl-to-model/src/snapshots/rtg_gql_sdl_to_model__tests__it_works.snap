---
source: gql-sdl-to-model/src/lib.rs
expression: model

---
V1 {
    entities: [
        DatabaseTable {
            name: "toto",
            sql_schema_name: "toto",
            sql_table_name: "toto",
            graphql_entity_type_name: "toto",
            graphql_filter_type_name: "toto",
            graphql_get_single_operation_name: "toto",
            graphql_get_list_operation_name: "toto",
            graphql_get_connection_operation_name: "toto",
            graphql_default_order_by: "toto",
            graphql_default_first: 0,
            graphql_default_offset: 10,
            fields: [
                ScalarDatabaseColumn {
                    name: "a",
                    sql_type: Text,
                    sql_column_name: "a",
                    graphql_field_name: "a",
                    graphql_type_name: "a",
                    graphql_order_by_asc: "a",
                    graphql_order_by_desc: "a",
                },
                ScalarDatabaseColumn {
                    name: "b",
                    sql_type: Text,
                    sql_column_name: "b",
                    graphql_field_name: "b",
                    graphql_type_name: "b",
                    graphql_order_by_asc: "b",
                    graphql_order_by_desc: "b",
                },
            ],
        },
    ],
}
